profiling/perfetto-connector/libperfetto-connector.cc

```diff
@@ -0,0 +1,110 @@
+#include "perfetto.h"
+#include <unistd.h>
+#include <fcntl.h>
+extern "C" void kokkosp_init_library(const int loadSeq,
+                                     const uint64_t interfaceVer,
+                                     const uint32_t,
+                                     void*);
+extern "C" void kokkosp_finalize_library();
+extern "C" void kokkosp_begin_parallel_for(const char* name,
+                                           const uint32_t,
+                                           uint64_t*);
+extern "C" void kokkosp_end_parallel_for(const uint64_t);
+extern "C" void kokkosp_begin_parallel_scan(const char* name,
+                                            const uint32_t,
+                                            uint64_t*);
+extern "C" void kokkosp_end_parallel_scan(const uint64_t);
+extern "C" void kokkosp_begin_parallel_reduce(const char* name,
+                                              const uint32_t,
+                                              uint64_t*);
+extern "C" void kokkosp_end_parallel_reduce(const uint64_t);
+extern "C" void kokkosp_push_profile_region(
+    char* regionName);
+extern "C" void kokkosp_pop_profile_region();
+
+
+PERFETTO_DEFINE_CATEGORIES(
+    perfetto::Category("kokkos.kernels")
+        .SetDescription("Kokkos Kernel Events"),
+    perfetto::Category("kokkos.regions").SetDescription(
+        "Kokkos Region Events"));
+PERFETTO_TRACK_EVENT_STATIC_STORAGE();
+perfetto::TraceConfig cfg;
+std::unique_ptr<perfetto::TracingSession> tracing_session;
+int fd;
```

Comment:
I would prefer if you prefixed these global variables with something like `kokkosp_perfetto_`

Reply:
Actually `cfg` is being shadowed in `kokkosp_init_library` which is the only place that was using a trace config in the first place.  Remove the global variable.